### ISSUE TITLE
Weird Training Loss plot for Training Task

### ISSUE THREAD

**Issue by Soothysay** [Optimization-AI/LibAUC#42]
Hi,
I am working on an extremely imbalanced binary classification problem where I follow the tutorial on the docs to use AUCMLoss with PESG optimizer instead of binary cross entropy loss and ADAM which I would use if I did not use LibAUC for my task. Surprisingly, I notice that the training loss increases with LibAUC and then drops.
![Training_Loss_True_LIBAUC](https://github.com/Optimization-AI/LibAUC/assets/63670764/e90485cb-341b-4a1e-ab51-59841c79e6b9)
However, for Binary Cross-Entropy + ADAM, the train loss variation pattern is as follows:
![Training_Loss_True_v4](https://github.com/Optimization-AI/LibAUC/assets/63670764/f37a446d-f75c-4a14-8e35-bb5e5445c5bb)

This seems surprising. Did you face any issue of this sort for any task? I think any insight on that might help me dig deep into this issue.

Thanks!

**Comment by GangLii** [comment]
Thanks for your interest in our library. This curve is a little unnormal. We didn't see it before. It might be caused by the extremely imbalanced dataset as you mentioned.  If you don't use a pretrained model for AUCMLoss , the extremely imbalanced dataset might cause this phenomenon if your random initialized solution may be not too bad. In this case, we would suggest that you could try another run with a different random seed to see if this phenomenon still exists. And changing the learning rate should be considered.

**Comment by optmai** [comment]
Hi, 

The loss produced from running the library (including Pytorch) is misleading. There are for mini-batch data. It does not say anything about the model quality. It is just a means to monitor the progress. 

I would suggest you to plot the AUROC or AP directly. You can also try to use our SOAP for AP maximization. 

Please let us know if this help address your issue.
